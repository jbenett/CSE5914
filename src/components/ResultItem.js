import React, { Component } from "react";
import PropTypes from "prop-types";
import Button from "./Button";

export default class ResultItem extends Component {
  static propTypes = {
    author: PropTypes.string,
    quote: PropTypes.string,
    original: PropTypes.string,
    image: PropTypes.string,
    id: PropTypes.string,
    removeSuggestion: PropTypes.func.isRequired,
    takeSuggestion: PropTypes.func.isRequired,
    onHoverSuggestion: PropTypes.func.isRequired
  };

  render() {
    return (
      <div
        className="result-item"
        onMouseEnter={() => {this.props.onHoverSuggestion(this.props.original);} }
        onMouseLeave={() => {this.props.onHoverSuggestion();} }
      >
        <div>
          <div className="result-options" onClick={()=>{this.props.removeSuggestion(this.props.id)} }>
            ✖️
          </div>
            <p className="result-item__replacement">{this.props.original}</p><span>&nbsp;➡️</span>
            <p className="result-item__replacing">{this.props.quote}</p>
          <Button onClick={()=>{ this.props.takeSuggestion(this.props.id)}}>
            Take Suggestion
          </Button>
          <div className="result-profile">
            <img
              className="profile-items profile-items--image"
              src={this.props.image}
            />
            <p className="profile-items profile-items--text">
              {this.props.author}
            </p>
          </div>
        </div>
      </div>
    );
  }
}
