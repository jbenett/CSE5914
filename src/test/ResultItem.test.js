import React from 'react';
import { shallow } from 'enzyme';
import ResultItem from '../components/ResultItem.js'

const myMock = jest.fn();
const clickMock =jest.fn();

describe('Result Item', () => {
  it('should render correctly', () => {
    const component = shallow(<ResultItem
      author="author"
      quote="text"
      contextPrefix=""
      original="source"
      image=""
      key="key"
      id="id"
      removeSuggestion={myMock}
      takeSuggestion={myMock}
      onHoverSuggestion={myMock}
    />);
  expect(component).toMatchSnapshot();
  });
  it('should close when the button is clicked', () => {
    const component = shallow(<ResultItem
      author="author"
      quote="text"
      contextPrefix=""
      original="source"
      image=""
      key="key"
      id="id"
      removeSuggestion={clickMock}
      takeSuggestion={myMock}
      onHoverSuggestion={myMock}
    />);
    component.find('.result-options').simulate('click');
    expect(clickMock).toHaveBeenCalled();
  })
});
