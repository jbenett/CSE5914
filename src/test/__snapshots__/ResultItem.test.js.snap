// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Item should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultItem
    author="author"
    contextPrefix=""
    id="id"
    image=""
    onHoverSuggestion={[MockFunction]}
    original="source"
    quote="text"
    removeSuggestion={[MockFunction]}
    takeSuggestion={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
          className="result-options"
          onClick={[Function]}
        >
          ✖️
        </div>
        <p
          className="result-item__replacement"
        >
          source
        </p>
        <span>
           ➡️
        </span>
        <p
          className="result-item__replacing"
        >
          text
        </p>
        <ForwardRef
          onClick={[Function]}
        >
          Take Suggestion
        </ForwardRef>
        <div
          className="result-profile"
        >
          <img
            className="profile-items profile-items--image"
            src=""
          />
          <p
            className="profile-items profile-items--text"
          >
            author
          </p>
        </div>
      </div>,
      "className": "result-item",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="result-options"
            onClick={[Function]}
          >
            ✖️
          </div>,
          <p
            className="result-item__replacement"
          >
            source
          </p>,
          <span>
             ➡️
          </span>,
          <p
            className="result-item__replacing"
          >
            text
          </p>,
          <ForwardRef
            onClick={[Function]}
          >
            Take Suggestion
          </ForwardRef>,
          <div
            className="result-profile"
          >
            <img
              className="profile-items profile-items--image"
              src=""
            />
            <p
              className="profile-items profile-items--text"
            >
              author
            </p>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "✖️",
            "className": "result-options",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "✖️",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "source",
            "className": "result-item__replacement",
          },
          "ref": null,
          "rendered": "source",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ➡️",
          },
          "ref": null,
          "rendered": " ➡️",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "text",
            "className": "result-item__replacing",
          },
          "ref": null,
          "rendered": "text",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Take Suggestion",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Take Suggestion",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "rules": Array [
                "
    display: inline-block;
    padding: 0.5rem 0.75rem;
    background-color: white;
    border-radius: 3px;
    font-family: ",
                [Function],
                ";
    font-weight: bold;
    transition: 150ms all;
    ",
                [Function],
                "

    &:hover {
        cursor: pointer;
        color: white;
        background-color: ",
                [Function],
                ";
        ",
                [Function],
                "
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="profile-items profile-items--image"
                src=""
              />,
              <p
                className="profile-items profile-items--text"
              >
                author
              </p>,
            ],
            "className": "result-profile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "profile-items profile-items--image",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "author",
                "className": "profile-items profile-items--text",
              },
              "ref": null,
              "rendered": "author",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
            className="result-options"
            onClick={[Function]}
          >
            ✖️
          </div>
          <p
            className="result-item__replacement"
          >
            source
          </p>
          <span>
             ➡️
          </span>
          <p
            className="result-item__replacing"
          >
            text
          </p>
          <ForwardRef
            onClick={[Function]}
          >
            Take Suggestion
          </ForwardRef>
          <div
            className="result-profile"
          >
            <img
              className="profile-items profile-items--image"
              src=""
            />
            <p
              className="profile-items profile-items--text"
            >
              author
            </p>
          </div>
        </div>,
        "className": "result-item",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="result-options"
              onClick={[Function]}
            >
              ✖️
            </div>,
            <p
              className="result-item__replacement"
            >
              source
            </p>,
            <span>
               ➡️
            </span>,
            <p
              className="result-item__replacing"
            >
              text
            </p>,
            <ForwardRef
              onClick={[Function]}
            >
              Take Suggestion
            </ForwardRef>,
            <div
              className="result-profile"
            >
              <img
                className="profile-items profile-items--image"
                src=""
              />
              <p
                className="profile-items profile-items--text"
              >
                author
              </p>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "✖️",
              "className": "result-options",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "✖️",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "source",
              "className": "result-item__replacement",
            },
            "ref": null,
            "rendered": "source",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " ➡️",
            },
            "ref": null,
            "rendered": " ➡️",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "text",
              "className": "result-item__replacing",
            },
            "ref": null,
            "rendered": "text",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Take Suggestion",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Take Suggestion",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "rules": Array [
                  "
    display: inline-block;
    padding: 0.5rem 0.75rem;
    background-color: white;
    border-radius: 3px;
    font-family: ",
                  [Function],
                  ";
    font-weight: bold;
    transition: 150ms all;
    ",
                  [Function],
                  "

    &:hover {
        cursor: pointer;
        color: white;
        background-color: ",
                  [Function],
                  ";
        ",
                  [Function],
                  "
    }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  className="profile-items profile-items--image"
                  src=""
                />,
                <p
                  className="profile-items profile-items--text"
                >
                  author
                </p>,
              ],
              "className": "result-profile",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "profile-items profile-items--image",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "author",
                  "className": "profile-items profile-items--text",
                },
                "ref": null,
                "rendered": "author",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
